e2e/upgrade-testing-cephfs.go

```diff
@@ -0,0 +1,109 @@
+package e2e
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo" // nolint
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = Describe("CephFS Upgrade Testing", func() {
+	f := framework.NewDefaultFramework("upgrade-test-cephfs")
+	var c clientset.Interface
+	// deploy cephfs CSI
+	BeforeEach(func() {
+		if !upgradeTesting || !testCephFS {
+			Skip("Skipping CephFS Upgrade Test")
+		}
+		c = f.ClientSet
+		if cephCSINamespace != defaultNs {
+			err = createNamespace(c, cephCSINamespace)
+			if err != nil {
+				Fail(err.Error())
+			}
+		}
+
+		// fetch current working directory to switch back
+		// when we are done upgrading.
+		cwd, err = os.Getwd()
+		if err != nil {
+			Fail(err.Error())
+		}
+		err = upgradeAndDeployCSI(upgradeVersion, "cephfs")
+		if err != nil {
+			Fail(err.Error())
+		}
+		createConfigMap(cephfsDirPath, f.ClientSet, f)
+		createCephfsSecret(f.ClientSet, f)
+		createCephfsStorageClass(f.ClientSet, f, true, "")
+	})
+	AfterEach(func() {
+		if !testCephFS || upgradeTesting {
+			Skip("Skipping CephFS Upgrade Test")
+		}
+		if CurrentGinkgoTestDescription().Failed {
+			// log pods created by helm chart
+			logsCSIPods("app=ceph-csi-cephfs", c)
+			// log provisoner
+			logsCSIPods("app=csi-cephfsplugin-provisioner", c)
+			// log node plugin
+			logsCSIPods("app=csi-cephfsplugin", c)
+		}
+		deleteConfigMap(cephfsDirPath)
+		deleteResource(cephfsExamplePath + "secret.yaml")
+		deleteResource(cephfsExamplePath + "storageclass.yaml")
+		if deployCephFS {
+			deleteCephfsPlugin()
+			if cephCSINamespace != defaultNs {
+				err = deleteNamespace(c, cephCSINamespace)
+				if err != nil {
+					Fail(err.Error())
+				}
+			}
+		}
+	})
+
+	Context("Cephfs Upgrade Test", func() {
+		It("Cephfs Upgrade Test", func() {
+
+			By("checking provisioner deployment is running")
+			err = waitForDeploymentComplete(cephfsDeploymentName, cephCSINamespace, f.ClientSet, deployTimeout)
+			if err != nil {
+				Fail(err.Error())
+			}
+
+			By("checking nodeplugin deamonsets is running")
+			err = waitForDaemonSets(cephfsDeamonSetName, cephCSINamespace, f.ClientSet, deployTimeout)
+			if err != nil {
+				Fail(err.Error())
+			}
+
+			By("upgrade to latest changes and verify app re-mount", func() {
+				pvcPath := cephfsExamplePath + "pvc.yaml"
+				appPath := cephfsExamplePath + "pod.yaml"
+				_, app := createPVCAndAppBinding(pvcPath, appPath, f, deployTimeout)
+				err = deletePod(app.Name, app.Namespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				deleteCephfsPlugin()
+
+				// support of PSP is available in current master, so we don't need to
+				// skip it now.
+				skipPSP = false
```

Comment:
remove this PSP check as we always going to have it as the minimum version we are testing is 2.1.2